Build fails/incomplete on requirements.txt schema not followed correctly

Since https://github.com/python-poetry/poetry-plugin-export/pull/258 was merged, the temporary requirements.txt format's changed. This makes local project dependencies unparseable. The build works, but the pex entrypoints do not exist / changed.

**requirements.txt with poetry-plugin-export@1.7.1**
```
--index-url https://gitlab-ci-token:[MASKED]@[MASKED].com/api/v4/projects/13599/packages/pypi/simple

alabaster==0.7.16 ; python_version >= "3.9" and python_version < "3.10"
annotated-types==0.6.0 ; python_version >= "3.9" and python_version < "3.10"
anyascii==0.3.2 ; python_version >= "3.9" and python_version < "3.10"
astroid==3.2.0 ; python_version >= "3.9" and python_version < "3.10"
asttokens==2.4.1 ; python_version >= "3.9" and python_version < "3.10"
autodoc-pydantic==2.2.0 ; python_version >= "3.9" and python_version < "3.10"
babel==2.15.0 ; python_version >= "3.9" and python_version < "3.10"
bda-base-dependencies @ file:///builds/cxa/cxa-ck/spark-sources/bda_base_dependencies ; python_version >= "3.9" and python_version < "3.10"
better-exceptions==0.3.3 ; python_version >= "3.9" and python_version < "3.10"
blinker==1.8.2 ; python_version >= "3.9" and python_version < "3.10"
bokeh==3.4.1 ; python_version >= "3.9" and python_version < "3.10"
certifi==2024.2.2 ; python_version >= "3.9" and python_version < "3.10"
cfgv==3.4.0 ; python_version >= "3.9" and python_version < "3.10"
charset-normalizer==3.3.2 ; python_version >= "3.9" and python_version < "3.10"
ck-spark-sources @ file:///builds/cxa/cxa-ck/spark-sources/sources ; python_version >= "3.9" and python_version < "3.10"
click==8.1.7 ; python_version >= "3.9" and python_version < "3.10"
colorama==0.4.6 ; python_version >= "3.9" and python_version < "3.10" and (sys_platform == "win32" or platform_system == "Windows")
contourpy==1.2.1 ; python_version >= "3.9" and python_version < "3.10"
coverage[toml]==7.5.1 ; python_version >= "3.9" and python_version < "3.10"
decorator==5.1.1 ; python_version >= "3.9" and python_version < "3.10"
distlib==0.3.8 ; python_version >= "3.9" and python_version < "3.10"
dm-knox-token==2.0.18 ; python_version >= "3.9" and python_version < "3.10"
dm-livy-submit==2.3.12 ; python_version >= "3.9" and python_version < "3.10"
dm-webhdfs==9.1.17 ; python_version >= "3.9" and python_version < "3.10"
docutils==0.20.1 ; python_version >= "3.9" and python_version < "3.10"
exceptiongroup==1.2.1 ; python_version >= "3.9" and python_version < "3.10"
executing==2.0.1 ; python_version >= "3.9" and python_version < "3.10"
filelock==3.14.0 ; python_version >= "3.9" and python_version < "3.10"
flask==3.0.3 ; python_version >= "3.9" and python_version < "3.10"
freezegun==1.5.1 ; python_version >= "3.9" and python_version < "3.10"
fsspec==2024.3.1 ; python_version >= "3.9" and python_version < "3.10"
identify==2.5.36 ; python_version >= "3.9" and python_version < "3.10"
idna==3.7 ; python_version >= "3.9" and python_version < "3.10"
imagesize==1.4.1 ; python_version >= "3.9" and python_version < "3.10"
importlib-metadata==7.1.0 ; python_version >= "3.9" and python_version < "3.10"
iniconfig==2.0.0 ; python_version >= "3.9" and python_version < "3.10"
ipdb==0.13.13 ; python_version >= "3.9" and python_version < "3.10"
ipython==8.18.1 ; python_version >= "3.9" and python_version < "3.10"
itsdangerous==2.2.0 ; python_version >= "3.9" and python_version < "3.10"
jedi==0.19.1 ; python_version >= "3.9" and python_version < "3.10"
jinja2==3.1.4 ; python_version >= "3.9" and python_version < "3.10"
json2html==1.3.0 ; python_version >= "3.9" and python_version < "3.10"
loguru==0.7.2 ; python_version >= "3.9" and python_version < "3.10"
markdown-it-py==3.0.0 ; python_version >= "3.9" and python_version < "3.10"
markupsafe==2.1.5 ; python_version >= "3.9" and python_version < "3.10"
matplotlib-inline==0.1.7 ; python_version >= "3.9" and python_version < "3.10"
mdit-py-plugins==0.4.1 ; python_version >= "3.9" and python_version < "3.10"
mdurl==0.1.2 ; python_version >= "3.9" and python_version < "3.10"
myst-parser==3.0.1 ; python_version >= "3.9" and python_version < "3.10"
nodeenv==1.8.0 ; python_version >= "3.9" and python_version < "3.10"
numpy==1.26.4 ; python_version >= "3.9" and python_version < "3.10"
packaging==24.0 ; python_version >= "3.9" and python_version < "3.10"
pandas==2.2.2 ; python_version >= "3.9" and python_version < "3.10"
parso==0.8.4 ; python_version >= "3.9" and python_version < "3.10"
pendulum==3.0.0 ; python_version >= "3.9" and python_version < "3.10"
pexpect==4.9.0 ; python_version >= "3.9" and python_version < "3.10" and sys_platform != "win32"
pillow==10.3.0 ; python_version >= "3.9" and python_version < "3.10"
platformdirs==4.2.1 ; python_version >= "3.9" and python_version < "3.10"
pluggy==1.5.0 ; python_version >= "3.9" and python_version < "3.10"
pre-commit==3.7.1 ; python_version >= "3.9" and python_version < "3.10"
prompt-toolkit==3.0.43 ; python_version >= "3.9" and python_version < "3.10"
ptyprocess==0.7.0 ; python_version >= "3.9" and python_version < "3.10" and sys_platform != "win32"
pure-eval==0.2.2 ; python_version >= "3.9" and python_version < "3.10"
py4j==0.10.9.5 ; python_version >= "3.9" and python_version < "3.10"
pydantic-core==2.18.2 ; python_version >= "3.9" and python_version < "3.10"
pydantic-settings==2.2.1 ; python_version >= "3.9" and python_version < "3.10"
pydantic==2.7.1 ; python_version >= "3.9" and python_version < "3.10"
pygments==2.18.0 ; python_version >= "3.9" and python_version < "3.10"
pyspark==3.3.0 ; python_version >= "3.9" and python_version < "3.10"
pytest-check==2.3.1 ; python_version >= "3.9" and python_version < "3.10"
pytest-cov==5.0.0 ; python_version >= "3.9" and python_version < "3.10"
pytest-flaskserver==1.2.10 ; python_version >= "3.9" and python_version < "3.10"
pytest==8.2.0 ; python_version >= "3.9" and python_version < "3.10"
python-dateutil==2.9.0.post0 ; python_version >= "3.9" and python_version < "3.10"
python-dotenv==1.0.1 ; python_version >= "3.9" and python_version < "3.10"
pytz==2022.7.1 ; python_version >= "3.9" and python_version < "3.10"
pyxattr==0.8.1 ; python_version >= "3.9" and python_version < "3.10"
pyyaml==6.0.1 ; python_version >= "3.9" and python_version < "3.10"
requests==2.31.0 ; python_version >= "3.9" and python_version < "3.10"
rich==13.7.1 ; python_version >= "3.9" and python_version < "3.10"
setuptools==69.5.1 ; python_version >= "3.9" and python_version < "3.10"
six==1.16.0 ; python_version >= "3.9" and python_version < "3.10"
snowballstemmer==2.2.0 ; python_version >= "3.9" and python_version < "3.10"
sphinx-autoapi==3.0.0 ; python_version >= "3.9" and python_version < "3.10"
sphinx-rtd-theme==2.0.0 ; python_version >= "3.9" and python_version < "3.10"
sphinx==7.3.7 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-applehelp==1.0.8 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-devhelp==1.0.6 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-htmlhelp==2.0.5 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-jquery==4.1 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-jsmath==1.0.1 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-qthelp==1.0.7 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-serializinghtml==1.1.10 ; python_version >= "3.9" and python_version < "3.10"
stack-data==0.6.3 ; python_version >= "3.9" and python_version < "3.10"
table-builder @ file:///builds/cxa/cxa-ck/spark-sources/table_builder ; python_version >= "3.9" and python_version < "3.10"
time-machine==2.14.1 ; python_version >= "3.9" and python_version < "3.10" and implementation_name != "pypy"
tomli==2.0.1 ; python_version >= "3.9" and python_version < "3.10"
tornado==6.4 ; python_version >= "3.9" and python_version < "3.10"
traitlets==5.14.3 ; python_version >= "3.9" and python_version < "3.10"
typing-extensions==4.11.0 ; python_version >= "3.9" and python_version < "3.10"
tzdata==2024.1 ; python_version >= "3.9" and python_version < "3.10"
urllib3==2.2.1 ; python_version >= "3.9" and python_version < "3.10"
utils @ file:///builds/cxa/cxa-ck/spark-sources/utils ; python_version >= "3.9" and python_version < "3.10"
virtualenv==20.26.1 ; python_version >= "3.9" and python_version < "3.10"
wcwidth==0.2.13 ; python_version >= "3.9" and python_version < "3.10"
werkzeug==3.0.3 ; python_version >= "3.9" and python_version < "3.10"
win32-setctime==1.1.0 ; python_version >= "3.9" and python_version < "3.10" and sys_platform == "win32"
xyzservices==2024.4.0 ; python_version >= "3.9" and python_version < "3.10"
zipp==3.18.1 ; python_version >= "3.9" and python_version < "3.10"
```

**requirements.txt with poetry-plugin-export@1.8.0**
```
--index-url https://gitlab-ci-token:[MASKED]@[MASKED].com/api/v4/projects/13599/packages/pypi/simple

-e file:///builds/cxa/cxa-ck/spark-sources/bda_base_dependencies ; python_version >= "3.9" and python_version < "3.10"
-e file:///builds/cxa/cxa-ck/spark-sources/sources ; python_version >= "3.9" and python_version < "3.10"
-e file:///builds/cxa/cxa-ck/spark-sources/table_builder ; python_version >= "3.9" and python_version < "3.10"
-e file:///builds/cxa/cxa-ck/spark-sources/utils ; python_version >= "3.9" and python_version < "3.10"
alabaster==0.7.16 ; python_version >= "3.9" and python_version < "3.10"
annotated-types==0.6.0 ; python_version >= "3.9" and python_version < "3.10"
anyascii==0.3.2 ; python_version >= "3.9" and python_version < "3.10"
astroid==3.2.0 ; python_version >= "3.9" and python_version < "3.10"
asttokens==2.4.1 ; python_version >= "3.9" and python_version < "3.10"
autodoc-pydantic==2.2.0 ; python_version >= "3.9" and python_version < "3.10"
babel==2.15.0 ; python_version >= "3.9" and python_version < "3.10"
better-exceptions==0.3.3 ; python_version >= "3.9" and python_version < "3.10"
blinker==1.8.2 ; python_version >= "3.9" and python_version < "3.10"
bokeh==3.4.1 ; python_version >= "3.9" and python_version < "3.10"
certifi==2024.2.2 ; python_version >= "3.9" and python_version < "3.10"
cfgv==3.4.0 ; python_version >= "3.9" and python_version < "3.10"
charset-normalizer==3.3.2 ; python_version >= "3.9" and python_version < "3.10"
click==8.1.7 ; python_version >= "3.9" and python_version < "3.10"
colorama==0.4.6 ; python_version >= "3.9" and python_version < "3.10" and (sys_platform == "win32" or platform_system == "Windows")
contourpy==1.2.1 ; python_version >= "3.9" and python_version < "3.10"
coverage[toml]==7.5.1 ; python_version >= "3.9" and python_version < "3.10"
decorator==5.1.1 ; python_version >= "3.9" and python_version < "3.10"
distlib==0.3.8 ; python_version >= "3.9" and python_version < "3.10"
dm-knox-token==2.0.18 ; python_version >= "3.9" and python_version < "3.10"
dm-livy-submit==2.3.12 ; python_version >= "3.9" and python_version < "3.10"
dm-webhdfs==9.1.16 ; python_version >= "3.9" and python_version < "3.10"
docutils==0.20.1 ; python_version >= "3.9" and python_version < "3.10"
exceptiongroup==1.2.1 ; python_version >= "3.9" and python_version < "3.10"
executing==2.0.1 ; python_version >= "3.9" and python_version < "3.10"
filelock==3.14.0 ; python_version >= "3.9" and python_version < "3.10"
flask==3.0.3 ; python_version >= "3.9" and python_version < "3.10"
freezegun==1.5.1 ; python_version >= "3.9" and python_version < "3.10"
fsspec==2024.3.1 ; python_version >= "3.9" and python_version < "3.10"
identify==2.5.36 ; python_version >= "3.9" and python_version < "3.10"
idna==3.7 ; python_version >= "3.9" and python_version < "3.10"
imagesize==1.4.1 ; python_version >= "3.9" and python_version < "3.10"
importlib-metadata==7.1.0 ; python_version >= "3.9" and python_version < "3.10"
iniconfig==2.0.0 ; python_version >= "3.9" and python_version < "3.10"
ipdb==0.13.13 ; python_version >= "3.9" and python_version < "3.10"
ipython==8.18.1 ; python_version >= "3.9" and python_version < "3.10"
itsdangerous==2.2.0 ; python_version >= "3.9" and python_version < "3.10"
jedi==0.19.1 ; python_version >= "3.9" and python_version < "3.10"
jinja2==3.1.4 ; python_version >= "3.9" and python_version < "3.10"
json2html==1.3.0 ; python_version >= "3.9" and python_version < "3.10"
loguru==0.7.2 ; python_version >= "3.9" and python_version < "3.10"
markdown-it-py==3.0.0 ; python_version >= "3.9" and python_version < "3.10"
markupsafe==2.1.5 ; python_version >= "3.9" and python_version < "3.10"
matplotlib-inline==0.1.7 ; python_version >= "3.9" and python_version < "3.10"
mdit-py-plugins==0.4.0 ; python_version >= "3.9" and python_version < "3.10"
mdurl==0.1.2 ; python_version >= "3.9" and python_version < "3.10"
myst-parser==3.0.1 ; python_version >= "3.9" and python_version < "3.10"
nodeenv==1.8.0 ; python_version >= "3.9" and python_version < "3.10"
numpy==1.26.4 ; python_version >= "3.9" and python_version < "3.10"
packaging==24.0 ; python_version >= "3.9" and python_version < "3.10"
pandas==2.2.2 ; python_version >= "3.9" and python_version < "3.10"
parso==0.8.4 ; python_version >= "3.9" and python_version < "3.10"
pendulum==3.0.0 ; python_version >= "3.9" and python_version < "3.10"
pexpect==4.9.0 ; python_version >= "3.9" and python_version < "3.10" and sys_platform != "win32"
pillow==10.3.0 ; python_version >= "3.9" and python_version < "3.10"
platformdirs==4.2.1 ; python_version >= "3.9" and python_version < "3.10"
pluggy==1.5.0 ; python_version >= "3.9" and python_version < "3.10"
pre-commit==3.7.0 ; python_version >= "3.9" and python_version < "3.10"
prompt-toolkit==3.0.43 ; python_version >= "3.9" and python_version < "3.10"
ptyprocess==0.7.0 ; python_version >= "3.9" and python_version < "3.10" and sys_platform != "win32"
pure-eval==0.2.2 ; python_version >= "3.9" and python_version < "3.10"
py4j==0.10.9.5 ; python_version >= "3.9" and python_version < "3.10"
pydantic-core==2.18.2 ; python_version >= "3.9" and python_version < "3.10"
pydantic-settings==2.2.1 ; python_version >= "3.9" and python_version < "3.10"
pydantic==2.7.1 ; python_version >= "3.9" and python_version < "3.10"
pygments==2.18.0 ; python_version >= "3.9" and python_version < "3.10"
pyspark==3.3.0 ; python_version >= "3.9" and python_version < "3.10"
pytest-check==2.3.1 ; python_version >= "3.9" and python_version < "3.10"
pytest-cov==5.0.0 ; python_version >= "3.9" and python_version < "3.10"
pytest-flaskserver==1.2.10 ; python_version >= "3.9" and python_version < "3.10"
pytest==8.2.0 ; python_version >= "3.9" and python_version < "3.10"
python-dateutil==2.9.0.post0 ; python_version >= "3.9" and python_version < "3.10"
python-dotenv==1.0.1 ; python_version >= "3.9" and python_version < "3.10"
pytz==2022.7.1 ; python_version >= "3.9" and python_version < "3.10"
pyxattr==0.8.1 ; python_version >= "3.9" and python_version < "3.10"
pyyaml==6.0.1 ; python_version >= "3.9" and python_version < "3.10"
requests==2.31.0 ; python_version >= "3.9" and python_version < "3.10"
rich==13.7.1 ; python_version >= "3.9" and python_version < "3.10"
setuptools==69.5.1 ; python_version >= "3.9" and python_version < "3.10"
six==1.16.0 ; python_version >= "3.9" and python_version < "3.10"
snowballstemmer==2.2.0 ; python_version >= "3.9" and python_version < "3.10"
sphinx-autoapi==3.0.0 ; python_version >= "3.9" and python_version < "3.10"
sphinx-rtd-theme==2.0.0 ; python_version >= "3.9" and python_version < "3.10"
sphinx==7.3.7 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-applehelp==1.0.8 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-devhelp==1.0.6 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-htmlhelp==2.0.5 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-jquery==4.1 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-jsmath==1.0.1 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-qthelp==1.0.7 ; python_version >= "3.9" and python_version < "3.10"
sphinxcontrib-serializinghtml==1.1.10 ; python_version >= "3.9" and python_version < "3.10"
stack-data==0.6.3 ; python_version >= "3.9" and python_version < "3.10"
time-machine==2.14.1 ; python_version >= "3.9" and python_version < "3.10" and implementation_name != "pypy"
tomli==2.0.1 ; python_version >= "3.9" and python_version < "3.10"
tornado==6.4 ; python_version >= "3.9" and python_version < "3.10"
traitlets==5.14.3 ; python_version >= "3.9" and python_version < "3.10"
typing-extensions==4.11.0 ; python_version >= "3.9" and python_version < "3.10"
tzdata==2024.1 ; python_version >= "3.9" and python_version < "3.10"
urllib3==2.2.1 ; python_version >= "3.9" and python_version < "3.10"
virtualenv==20.26.1 ; python_version >= "3.9" and python_version < "3.10"
wcwidth==0.2.13 ; python_version >= "3.9" and python_version < "3.10"
werkzeug==3.0.3 ; python_version >= "3.9" and python_version < "3.10"
win32-setctime==1.1.0 ; python_version >= "3.9" and python_version < "3.10" and sys_platform == "win32"
xyzservices==2024.4.0 ; python_version >= "3.9" and python_version < "3.10"
zipp==3.18.1 ; python_version >= "3.9" and python_version < "3.10"
```

With the pex created based on the requierements.txt of v1.8.0, our pipeline failed:
```bash
$ cd tests/test_table_builder/
$ ../../bda_base_dependencies/$BDA_BASE_DEPENDENCIES_DIST_WITH_SPARK_VERSION/bda-base-dependencies.pex -m pytest -m 'not system_test_only' --junitxml=unit_tests_table_builder.xml --cov=table_builder --cov-report term --cov-report xml:unit_tests_table_builder_coverage.xml
ImportError while loading conftest '/builds/cxa/cxa-ck/spark-sources/tests/test_table_builder/conftest.py'.
conftest.py:4: in <module>
    from test_helpers.hive_fixtures import add_database, add_managed_table 
E   ModuleNotFoundError: No module named 'test_helpers'
```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,14 @@
 # Release Notes
 
+## 2.10.1
+
+This release fixes a long-standing bug in Pex parsing of editable
+requirements. This bug caused PEXes containing local editable project
+requirements to fail to import those local editable projects despite
+the fact the PEX itself contained them.
+
+* Fix editable requirement parsing. (#2464)
+
 ## 2.10.0
 
 This release adds support for injecting requirements into the isolated

--- a/docs-requirements.txt
+++ b/docs-requirements.txt
@@ -3,3 +3,7 @@ httpx
 myst-parser[linkify]
 sphinx
 sphinx-simplepdf
+
+# The 0.11.0 release removes deprecated API parameters which breaks weasyprint (62.3 depends on
+# `pydyf>=0.10.0`) which is a dependency of sphinx-simplepdf.
+pydyf<0.11.0

--- a/pex/requirements.py
+++ b/pex/requirements.py
@@ -652,7 +652,7 @@ def parse_requirements(
                         yield requirement
                 continue
 
-            # Skip empty lines, comment lines and all other Pip options.
+            # Skip empty lines, comment lines and all Pip global options.
             if not processed_text or (
                 processed_text.startswith("-")
                 and not re.match(r"^(?:-e|--editable)\s.*", processed_text)

--- a/pex/requirements.py
+++ b/pex/requirements.py
@@ -418,7 +418,9 @@ def _try_parse_pip_local_formats(
     directory_name, requirement_parts = match.groups()
     stripped_path = os.path.join(os.path.dirname(path), directory_name)
     abs_stripped_path = (
-        os.path.join(basepath, stripped_path) if basepath else os.path.abspath(stripped_path)
+        os.path.join(basepath, stripped_path)
+        if basepath and not os.path.isabs(stripped_path)
+        else os.path.abspath(stripped_path)
     )
     if not os.path.exists(abs_stripped_path):
         return None
@@ -651,7 +653,10 @@ def parse_requirements(
                 continue
 
             # Skip empty lines, comment lines and all other Pip options.
-            if not processed_text or processed_text.startswith("-"):
+            if not processed_text or (
+                processed_text.startswith("-")
+                and not re.match(r"^(?:-e|--editable)\s.*", processed_text)
+            ):
                 continue
 
             # Only requirement lines remain.

--- a/pex/version.py
+++ b/pex/version.py
@@ -1,4 +1,4 @@
 # Copyright 2015 Pex project contributors.
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-__version__ = "2.10.0"
+__version__ = "2.10.1"

--- a/tests/integration/test_issue_2410.py
+++ b/tests/integration/test_issue_2410.py
@@ -1,0 +1,77 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import os.path
+import subprocess
+from textwrap import dedent
+
+from colors import colors
+
+from pex.common import safe_open
+from pex.typing import TYPE_CHECKING
+from testing import run_pex_command
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def test_pex_with_editable(tmpdir):
+    # type: (Any) -> None
+
+    project_dir = os.path.join(str(tmpdir), "project")
+    with safe_open(os.path.join(project_dir, "example.py"), "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                import sys
+
+                import colors
+
+
+                def colorize(*messages):
+                    return colors.green(" ".join(messages))
+
+
+                if __name__ == "__main__":
+                    print(colorize(*sys.argv[1:]))
+                    sys.exit(0)
+                """
+            )
+        )
+    with safe_open(os.path.join(project_dir, "setup.py"), "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                from setuptools import setup
+
+
+                setup(
+                    name="example",
+                    version="0.1.0",
+                    py_modules=["example"],
+                )
+                """
+            )
+        )
+
+    requirements = os.path.join(project_dir, "requirements.txt")
+    with safe_open(requirements, "w") as fp:
+        fp.write(
+            dedent(
+                """\
+                ansicolors==1.1.8
+                -e file://{project_dir}
+                """
+            ).format(project_dir=project_dir)
+        )
+
+    pex = os.path.join(str(tmpdir), "pex")
+    run_pex_command(args=["-r", requirements, "-m", "example", "-o", pex]).assert_success()
+    output = (
+        subprocess.check_output(args=[pex, "A", "wet", "duck", "flies", "at", "night!"])
+        .decode("utf-8")
+        .strip()
+    )
+    assert colors.green("A wet duck flies at night!") == output, output

--- a/tests/test_requirements.py
+++ b/tests/test_requirements.py
@@ -51,7 +51,7 @@ def chroot():
         curdir = os.getcwd()
         try:
             os.chdir(chroot)
-            yield chroot
+            yield os.path.realpath(chroot)
         finally:
             os.chdir(curdir)
 

--- a/tests/test_requirements.py
+++ b/tests/test_requirements.py
@@ -278,8 +278,8 @@ def test_parse_requirements_stress(chroot):
                 hg+http://hg.example.com/MyProject@da39a3ee5e6b\\
                     #egg=AnotherProject[extra,more] ; python_version=="3.9.*"&subdirectory=foo/bar
 
-                ftp://a/${PROJECT_NAME}-1.0.tar.gz
-                http://a/${PROJECT_NAME}-1.0.zip
+                ftp://a/${{PROJECT_NAME}}-1.0.tar.gz
+                http://a/${{PROJECT_NAME}}-1.0.zip
                 https://a/numpy-1.9.2-cp34-none-win32.whl
                 https://a/numpy-1.9.2-cp34-none-win32.whl;\\
                     python_version=="3.4.*" and sys_platform=='win32'
@@ -295,8 +295,14 @@ def test_parse_requirements_stress(chroot):
 
                 # Wheel with local version
                 http://download.pytorch.org/whl/cpu/torch-1.12.1%2Bcpu-cp310-cp310-linux_x86_64.whl
+
+                # Editable
+                -e file://{chroot}/extra/a/local/project
+                --editable file://{chroot}/extra/a/local/project/
+                -e ./another/local/project
+                --editable ./another/local/project/
                 """
-            )
+            ).format(chroot=chroot)
         )
     touch("extra/pyproject.toml")
     touch("extra/a/local/project/pyproject.toml")
@@ -470,6 +476,10 @@ def test_parse_requirements_stress(chroot):
             url="http://download.pytorch.org/whl/cpu/torch-1.12.1%2Bcpu-cp310-cp310-linux_x86_64.whl",
             specifier="==1.12.1+cpu",
         ),
+        local_req(path=os.path.join(chroot, "extra/a/local/project"), editable=True),
+        local_req(path=os.path.join(chroot, "extra/a/local/project"), editable=True),
+        local_req(path=os.path.join(chroot, "extra/another/local/project"), editable=True),
+        local_req(path=os.path.join(chroot, "extra/another/local/project"), editable=True),
         url_req(
             project_name="numpy",
             url=os.path.realpath("./downloads/numpy-1.9.2-cp34-none-win32.whl"),

